c_cxx/OpenVINO_EP/Linux/squeezenet_classification/README.md

```diff
@@ -37,26 +41,43 @@ If you build it by yourself, you must append the "--build_shared_lib" flag to yo
-```
-Note: This build command is using the opencv location from OpenVINO 2021.3 Release Installation. You can use any version of OpenVINO and change the location path accordingly.
+
+   - For general sample
+      ```
+      g++ -o run_squeezenet squeezenet_cpp_app.cpp -I ../../../include/onnxruntime/core/session/ -I /opt/intel/openvino_2021.4.752/opencv/include/ -I /opt/intel/openvino_2021.4.752/opencv/lib/ -L ./ -lonnxruntime_providers_openvino -lonnxruntime_providers_shared -lonnxruntime -L /opt/intel/openvino_2021.4.752/opencv/lib/ -lopencv_imgcodecs -lopencv_dnn -lopencv_core -lopencv_imgproc
+      ```
+      Note: This build command is using the opencv location from OpenVINO 2021.4.1 Release Installation. You can use any version of OpenVINO and change the location path accordingly.
+
+   - For the sample using IO Buffer Optimization feature
+      Set the OpenCL lib and headers path. For example if you are setting the path from openvino source build folder, the paths will be like:
+      ```
+      export OPENCL_LIBS=path/to/your/directory/openvino/bin/intel64/Debug/lib/
+      export OPENCL_INCS=path/to/your/directory/openvino/thirdparty/ocl/clhpp_headers/include/
+      ```
+      Now based on the above path, compile command will be:
+      ```
+      g++ -o run_squeezenet squeezenet_cpp_app_io.cpp -I ../../../include/onnxruntime/core/session/ -I $OPENCL_INCS -I $OPENCL_INCS/../../cl_headers/ -I /opt/intel/openvino_2021.4.752/opencv/include/ -I /opt/intel/openvino_2021.4.752/opencv/lib/ -L ./ -lonnxruntime_providers_openvino -lonnxruntime_providers_shared -lonnxruntime -L /opt/intel/openvino_2021.4.752/opencv/lib/ -lopencv_imgcodecs -lopencv_dnn -lopencv_core -lopencv_imgproc -L $OPENCL_LIBS -lOpenCL
+      ```
 
 4. Run the sample
 
-To Run
-(using Intel OpenVINO-EP)
-```
-./run_squeezenet --use_openvino <path_to_onnx_model> <path_to_sample_image> <path_to_labels_file>
-```
-Example:
-```
-./run_squeezenet --use_openvino squeezenet1.1-7.onnx demo.jpeg synset.txt  (using Intel OpenVINO-EP)
-```
-(using Default CPU)
-```
-./run_squeezenet --use_cpu <path_to_onnx_model> <path_to_sample_image> <path_to_labels_file>
-```
+   - To Run the general sample
+      (using Intel OpenVINO-EP)
+      ```
+      ./run_squeezenet --use_openvino <path_to_onnx_model> <path_to_sample_image> <path_to_labels_file>
+      ```
+      Example:
+      ```
+      ./run_squeezenet --use_openvino squeezenet1.1-7.onnx demo.jpeg synset.txt  (using Intel OpenVINO-EP)
+      ```
+      (using Default CPU)
+      ```
+      ./run_squeezenet --use_cpu <path_to_onnx_model> <path_to_sample_image> <path_to_labels_file>
+      ```
+   - To Run the sample for IO Buffer Optimization feature
```

Comment:
@mayavijx  this sample using IO Buffer will only work with OpenVINO and not MLAS. so thats why we are not passing any flags like --use_openvino or --use_cpu to run this sample. is my understanding correct and the sample directly runs on openvino.

Reply:
Yes Maajid